src/app/admin/login/login.component.html

```diff
@@ -0,0 +1,38 @@
+<form
+  fxLayout="column"
+  fxLayoutAlign="center center"
+  fxLayoutGap="10px"
+  [formGroup]="loginForm"
+  (ngSubmit)="submit()">
+  <h2>Login into Admin Panel</h2>
+  <mat-form-field appearance="fill">
+    <input
+      type="email"
+      matInput
+      placeholder="your email"
+      formControlName="email">
+    <mat-error *ngIf="email?.touched  " class="validation">
+      <small *ngIf="email?.errors?.['required']">Field email should not be empty</small>
```

Comment:
one HTML element, create the map <Error, ErrorMessage> and get the error message from map[errorr]